lib/services/discord_rpc.dart

```diff
@@ -0,0 +1,84 @@
+import 'dart:async';
+import 'dart:core';
+
+import 'package:finamp/services/media_state_stream.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:get_it/get_it.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+import 'package:flutter_discord_rpc/flutter_discord_rpc.dart';
+part 'discord_rpc.g.dart';
+
+var running = false;
+StreamSubscription<MediaState>? listener;
+
+@riverpod
+class DiscordRpc extends _$DiscordRpc {
+  @override
+  bool build() {
+    bool enabled = true;
+    return enabled;
+  }
+
+  static void startWatching() async {
+    ProviderContainer container = GetIt.instance<ProviderContainer>();
+    container.listen(discordRpcProvider, (_, enabled) {
+      enabled ? DiscordRpc.start() : DiscordRpc.stop(); 
+    });
+  }
+
+  static Future<void> start() async {
+    if (!running) {
+      running = true;
+      await FlutterDiscordRPC.initialize("1363567299948314906");
```

Comment:
you could use the unique device ID from the auth header for this?

Reply:
ah nevermind, this ID is coming from Discord.  
Is it a secret or can we just expose it like that?

Reply:
This is public, yes
When you go to discord.dev and create an application with the name "Jellyfin" (thats what I did) and then simply use the appID, the RPC will show "listening to Jellyfin"

Reply:
yes, only once
because Im currently reusing the id from my rpc xd

Reply:
did you need to configure the app in the Discord portal? or is the "listening" bit done just through the code?

Reply:
Yes its code. It can be listening, streaming, watching and playing